tests/dataset_runner.py

```diff
@@ -148,27 +185,120 @@ def complete_submission(self):
         if response.status_code != requests.codes.accepted:
             raise RuntimeError(f"PUT {submit_url} returned {response.status_code}: {response.content}")
         Progress.report("  done.\n")
-
-    def wait_for_primary_and_results_bundles(self):
+    
+    def wait_for_primary_bundles_analysis_workflows_and_results_bundles(self):
         """
-        We used to wait for all primary bundles to be created before starting to look for results.
+        We wait for all primary bundles to be created before starting to look for results in non-scaling tests.
         It appears that with large submissions, results can start to appear before bundle export is finished,
-        so we now monitor both kinds of bundles simultaneously.
+        so we monitor both kinds of bundles simultaneously in scaling tests.
         """
-        Progress.report("WAITING FOR PRIMARY AND RESULTS BUNDLE(s) TO BE CREATED...")
+        Progress.report("WAITING FOR PRIMARY BUNDLE(s), ANALYSIS WORKFLOWS AND RESULTS BUNDLE(s)...")
         self.expected_bundle_count = self.dataset.config["expected_bundle_count"]
         WaitFor(
-            self._count_primary_and_secondary_bundles
+            self._count_primary_bundles_analysis_workflows_and_results_bundles
         ).to_return_value(value=self.expected_bundle_count)
 
-    def _count_primary_and_secondary_bundles(self):
+    def _count_primary_bundles_analysis_workflows_and_results_bundles(self):
         if self._primary_bundle_count() < self.expected_bundle_count:
             self._count_primary_bundles()
+        if self._analysis_workflows_count() < self.expected_bundle_count:
+            self._batch_count_analysis_workflows_by_project_shortname()
         if self._results_bundles_count() < self.expected_bundle_count:
             self._count_results_bundles()
-        Progress.report("  bundles: primary: {}/{}, results: {}/{}".format(
+        Progress.report("  bundles: primary: {0}/{1}, results: {2}/{3} \n  workflows: running: {4}/{5}, \
```

Comment:
Minor suggestion: what do you think about putting the `results` at the end?

Reply:
Done!